instagram-app/src/components/PostContainer/PostPage.js

```diff
@@ -0,0 +1,132 @@
+import React, { Component } from 'react';
+//import withAuthenticate from './components/Authentication/withAuthenticate';
+import '../../App.css';
+import SearchBar from '../SearchBar/SearchBar';
+import dummyData from '../../dummy-data';
+import PostContainer from './PostContainer';
+
+class PostPage extends Component {
+  constructor() {
+    super();
+    this.state = {
+      data: [],
+      comments: {
+        text: '',
+        username: ''
+      }
+     
+    }
+  }
+  
+
+  addComment = (e) => {
+    e.preventDefault();
+    const thisIndex = e.target.getAttribute('index')
+    //console.log(this.state.data[thisIndex].comments)
+    this.state.data[thisIndex].comments.push({ text: this.state.comments.text, username: localStorage.getItem('user')})
+    this.setState({
+      data: [...this.state.data ]
+    })
+  }
+
+  handleChange = e => {
+    //console.log(e.target.value)
+    //this.state.data[0].comments.push({ text:e.target.value, username: "Billy Bob"})
+    this.setState({
+      data: [...this.state.data],
+      comments: {
+        text: e.target.value,
+        username: "BillyBob"
+      }
+    })
+  }
+
+ handleSearch = e => {
+    //console.log(e.target.value)
+    //this.setState({data: [...this.state.data]})
+    //const text = e.target.value.split("").toString()
+    //console.log(text)
+    //const newArray = this.state.data.filter(item => (this.setState({data: item})))
+    const newArray = []
+    this.state.data.filter(function(post) {
+      if (post.username === e.target.value) {
+       newArray.push(post)
+       console.log(newArray)
+       return newArray
+      } else {
+        return newArray
+      }
+      
+    });
+      
+        
+    if (newArray.length > 0) {
+      this.setState({data: newArray})
+    } else if (newArray.length === 0) {
+      this.setState({data: dummyData})
+    }
+  }
+ 
+  
+
+  handleLikes = e => {
+    const thisIndex = e.target.getAttribute('index')
+    //this.state.data[thisIndex].likes = this.state.data[thisIndex].likes + 1
+    console.log(e.target.getAttribute('class'))
+    e.target.classList.toggle('red-heart')
+      if (e.target.getAttribute('class') === 'heart red-heart') {
+      this.state.data[thisIndex].likes = this.state.data[thisIndex].likes + 1
+      this.setState({
+        //newLikes: this.state.data[thisIndex].likes + 1,
+        data: [...this.state.data]
```

Comment:
State values (like `this.state.data`) shouldn't be modified directly - it is best practice to use `.setState` instead for changes.

To avoid modifying the data state value directly, you can copy it to a new variable first and then use that new variable's value in this.setState, like so:
```
const newData = this.state.data.slice();
newData[thisIndex].likes = this.state.data[thisIndex].likes + 1;
this.setState({
    data: [...newData]
});
```